src/_guides/libraries/java-interop.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Java interop using package:jnigen"
+description: "To use Java in your Dart program, use package:jnigen."
+jnigen: "https://pub.dev/packages/jnigen"
+jni: "https://pub.dev/packages/jni"
+example: "https://github.com/HosseinYousefi/jnigen_example/tree/main"
+jnidoc: "https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/jniTOC.html"
+---
+
+Dart mobile, command-line, and server apps
+running on the [Dart Native platform](/overview#platform), on Android, Windows
+and Linux can use [`package:jni`]({{page.jni}})
+and [`package:jnigen`]({{page.jnigen}})
+to call Java and Kotlin APIs.
+
+{{site.alert.note}}
+  This interop feature is **experimental**,
+  and [in active development](https://github.com/dart-lang/jnigen/issues).
```

Comment:
Consider linking to an overarching issue such as https://github.com/dart-lang/sdk/issues/49674 so developer can follow the progress easier.